src/cmdstan/command.hpp

```diff
@@ -594,6 +594,149 @@ int command(int argc, const char *argv[]) {
     return_code = stan::services::standalone_generate(
         model, fitted_params.samples.block(0, meta_cols, num_rows, num_cols),
         random_seed, interrupt, logger, sample_writers[0]);
+  } else if (user_method->arg("log_prob")) {
+    string_argument *upars_file = dynamic_cast<string_argument *>(
+        parser.arg("method")->arg("log_prob")->arg("unconstrained_params"));
+    string_argument *cpars_file = dynamic_cast<string_argument *>(
+        parser.arg("method")->arg("log_prob")->arg("constrained_params"));
+    bool jacobian_adjust
+        = dynamic_cast<bool_argument *>(
+              parser.arg("method")->arg("log_prob")->arg("jacobian_adjust"))
+              ->value();
+    if (upars_file->is_default() && cpars_file->is_default()) {
+      msg << "No input parameters provided, cannot calculate log-probability";
+      throw std::invalid_argument(msg.str());
+    }
+
+    /*
+      By only updating the sizes/vectors if there is data present, the
+      calculation and printing can be agnostic from the combinations of
+      un-/constrained parameters passed
+    */
+    size_t u_params_vec_size = 0;
+    size_t u_params_size = 0;
+    size_t c_params_vec_size = 0;
+    size_t c_params_size = 0;
+
+    std::vector<double> u_params_r;
+    std::vector<double> c_params_r;
+    std::vector<size_t> dims_u_params_r;
+    std::vector<size_t> dims_c_params_r;
+    std::vector<int> dummy_params_i;
+
+    if (!(upars_file->is_default())) {
+      std::string u_fname(upars_file->value());
+      std::ifstream u_stream(u_fname.c_str());
+
+      std::shared_ptr<stan::io::var_context> upars_context
+          = get_var_context(u_fname);
+
+      u_params_r = (*upars_context).vals_r("params_r");
+      dims_u_params_r = (*upars_context).dims_r("params_r");
+
+      // Detect whether multiple sets of parameter values have been passed
+      // and set the sizes accordingly
+      u_params_vec_size = dims_u_params_r.size() == 2 ? dims_u_params_r[0] : 1;
+      u_params_size = dims_u_params_r.size() == 2 ? dims_u_params_r[1]
+                                                  : dims_u_params_r[0];
+    }
+
+    if (!(cpars_file->is_default())) {
+      std::string c_fname(cpars_file->value());
+      std::ifstream c_stream(c_fname.c_str());
+
+      std::shared_ptr<stan::io::var_context> cpars_context
+          = get_var_context(c_fname);
+
+      c_params_r = (*cpars_context).vals_r("params_r");
+      dims_c_params_r = (*cpars_context).dims_r("params_r");
+
+      c_params_vec_size = dims_c_params_r.size() == 2 ? dims_c_params_r[0] : 1;
+      c_params_size = dims_c_params_r.size() == 2 ? dims_c_params_r[1]
+                                                  : dims_c_params_r[0];
+    }
+
+    /*
+    Parameter names and dimensions required for constructing
+    array_var_context and subsequently transforming constrained parameters
+    to unconstrained scale
+    */
```

Comment:
shorten these doc strings---the names make it pretty clear already, and long doc strings get in the way of reading code.  For example, this one can be a one-liner:

```
store names and dims for constructing array_var_context and unconstraining transform
```

and the second one can be

```
store in single nested array to allow single loop for calc and print
```

The comments like this don't need to be long grammatical sentences---they're just for devs and the shorter the better.